src/drivers/vercel-kv.ts

```diff
@@ -0,0 +1,43 @@
+import { createClient } from "@vercel/kv";
+import type { RedisConfigNodejs } from "@upstash/redis";
+
+import { defineDriver } from "./utils";
+
+export interface RedisOptions extends RedisConfigNodejs {
+  base?: string;
+}
+
+export default defineDriver<RedisOptions>((opts) => {
+  const redis = createClient(opts);
+
+  let base = opts?.base || "";
+  if (base && !base.endsWith(":")) {
+    base += ":";
+  }
+  const r = (key: string) => base + key;
+
+  return {
+    hasItem(key) {
+      return redis.exists(r(key)).then(Boolean);
+    },
+    getItem(key) {
+      return redis.get(r(key));
+    },
+    setItem(key, value) {
+      return redis.set(r(key), value).then(() => {});
+    },
+    removeItem(key) {
+      return redis.del(r(key)).then(() => {});
+    },
+    getKeys() {
```

Comment:
Perf:We could implement getKeys with base while now it always makes a query for all keys and filters on client side

Reply:
Yes. And yes other driver needs fix